bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -490,12 +489,12 @@ public State getState(Widget w) {
     private State convertState(Widget w, State s) {
         State returnState = null;
 
-        if (w instanceof SwitchImpl) {
-            if (s instanceof PercentType) {
+        if (w instanceof Switch) {
+            if (s instanceof Convertible) {
```

Comment:
Why not just: `returnState = s.getStateAs(OnOffType.class);`
This call is doing the test with Covertible.